swarm.go

```diff
@@ -176,6 +177,20 @@ func (s *Swarm) teardown() error {
 	// Wait for everything to finish.
 	s.refs.Wait()
 
+	// Now close out any transports (if necessary).
+	s.transports.Lock()
+	transports := s.transports.m
+	s.transports.m = nil
+	s.transports.Unlock()
+
+	for t := range transports {
+		if closer, ok := t.(io.Closer); ok {
+			if err := closer.Close(); err != nil {
```

Comment:
In my case (tor and i2p) `closer.Close()` is slow, it needs to call other peers in the dht about him leaving, that takes some times, I would like each transport to have his own goroutine.

Reply:
Eh, I'd prefer to be able to _wait_ and I'm not too concerned about the cost of firing off goroutines on shutdown.